components/post_markdown/index.js

```diff
@@ -3,19 +3,29 @@
 
 import {connect} from 'react-redux';
 import {getChannel} from 'mattermost-redux/selectors/entities/channels';
+import {getAllUserMentionKeys} from 'mattermost-redux/selectors/entities/search';
+import {getCurrentUserMentionKeys} from 'mattermost-redux/selectors/entities/users';
+import {memoizeResult} from 'mattermost-redux/utils/helpers';
 
 import {canManageMembers} from 'utils/channel_utils.jsx';
 
 import PostMarkdown from './post_markdown';
 
 function mapStateToProps(state, ownProps) {
     const channel = getChannel(state, ownProps.channelId);
+    const allMentionKeys = getAllUserMentionKeys(state);
+    const mentionKeysWithoutChannelMentions = filterOutChannelMentions(allMentionKeys);
     return {
         channel,
         pluginHooks: state.plugins.components.MessageWillFormat,
         hasPluginTooltips: Boolean(state.plugins.components.LinkTooltip),
         isUserCanManageMembers: channel && canManageMembers(state, channel),
+        allMentionKeys,
+        mentionKeysWithoutGroups: getCurrentUserMentionKeys(state),
+        mentionKeysWithoutChannelMentions,
```

Comment:
Looks good to me but does it make sense maybe to check the `post.props` here and only send one set of mention keys to the component? thoughts?

Reply:
I think the way it currently is fine. I don't see moving the logic to `index.js` will give it any sort of improvement so I think keeping it the way it currently is should be okay.

Reply:
It would definitely be cleaner to do it that way since we'd have all of the logic for it in one place instead of having to pass multiple props through here and then do stuff with them later on. You could combine them all into one selector like

```javascript
function makeGetMentionKeysForPost() {
    return createSelector(
        getAllUserMentionKeys,
        getCurrentUserMentionKeys,
        (state, post) => post,
        (allMentionKeys, mentionKeysWithoutGroups, post) => {
            let mentionKeys = allMentionKeys;
            if (post?.props?.disable_group_highlight) {
                mentionKeys = mentionKeysWithoutGroups;
            }

            if (post?.props?.mentionHighlightDisabled) {
                mentionKeys = mentionKeys.filter((value) => !['@all', '@channel', '@here'].includes(value.key));
            }

            return mentionKeys;
        }
    );
}

function makeMapStateToProps() {
    const getMentionKeysForPost = makeGetMentionKeysForPost();

    return (state, ownProps) => {
        return {
            mentionKeys: getMentionKeysForPost(state, ownProps.post);
        };
    };
}
```
Note that this is using a selector factory. Since the selector takes an argument, each post needs its own version of the selector to not overwrite each others' memoized results.

This version should also fix the behaviour change I mentioned before.

Reply:
@hmhealey 
Ok I have refactored it and added test for that.